src/audio/aria/aria.c

```diff
@@ -0,0 +1,348 @@
+	const size_t chc = cd->base.audio_fmt.channels_count;
+	const size_t sgs = (cd->base.audio_fmt.valid_bit_depth >> 3) * chc;
+	const size_t sgc = ibs / sgs;
+	const size_t req_mem = GetRequiredMemory(chc, sgc);
+	size_t att = aria->attenuation_;
+
+	if (aria->attenuation_ > ARIA_MAX_ATT) {
+		comp_warn(dev, "init_aria(): Attenuation value %d must not be greater than %d",
+			  att, ARIA_MAX_ATT);
+		att = ARIA_MAX_ATT;
+	}
+
+	comp_set_drvdata(dev, cd);
+
+	void *buf = rballoc(0, SOF_MEM_CAPS_RAM, req_mem);
+	void *buf_in = rballoc(0, SOF_MEM_CAPS_RAM, req_mem);
+	void *buf_out = rballoc(0, SOF_MEM_CAPS_RAM, req_mem);
+
+	if (!buf || !buf_in || !buf_out) {
+		comp_err(dev, "init_aria(): allocation failed for size %d", req_mem);
+		return -ENOMEM;
+	}
+
+	ret = aria_algo_init(dev, buf, buf_in, buf_out, att, chc, sgc);
+	if (ret < 0)
+		return ret;
+
+	return 0;
+}
+
+static struct comp_dev *aria_new(const struct comp_driver *drv,
+				 struct comp_ipc_config *config, void *spec)
+{
+	struct comp_dev *dev;
+	int ret;
+
+	comp_cl_info(&comp_aria, "aria_new()");
+
+	dev = comp_alloc(drv, sizeof(*dev));
+	if (!dev)
+		return NULL;
+
+	ret = init_aria(dev, config, spec);
+	if (ret < 0) {
+		comp_free(dev);
+		return NULL;
+	}
+	dev->state = COMP_STATE_READY;
+
+	return dev;
+}
+
+static void aria_free(struct comp_dev *dev)
+{
+	struct aria_data *cd = comp_get_drvdata(dev);
+
+	rfree(cd->data_);
+	rfree(cd->buf_in);
+	rfree(cd->buf_out);
+	rfree(cd);
+	rfree(dev);
+}
+
+static int aria_prepare(struct comp_dev *dev)
+{
+	int ret;
+
+	comp_info(dev, "aria_prepare()");
+
+	if (dev->state == COMP_STATE_ACTIVE) {
+		comp_info(dev, "aria_prepare(): Component is in active state.");
+		return 0;
+	}
+
+	ret = comp_set_state(dev, COMP_TRIGGER_PREPARE);
+	if (ret < 0)
+		return ret;
+
+	if (ret == COMP_STATUS_STATE_ALREADY_SET)
+		return PPL_STATUS_PATH_STOP;
+
+	return 0;
+}
+
+static int aria_reset(struct comp_dev *dev)
+{
+	struct aria_data *cd = comp_get_drvdata(dev);
+	int idx;
+	int ret = 0;
+
+	comp_info(dev, "aria_reset()");
+
+	if (dev->state == COMP_STATE_ACTIVE) {
+		comp_info(dev, "aria module is in active state. Ignore resetting");
+		return 0;
+	}
+
+	for (idx = 0; idx < ARIA_MAX_GAIN_STATES; ++idx)
+		cd->gains_[idx] = (1ULL << (32 - cd->att_ - 1)) - 1;
+
+	memset(cd->data_, 0, sizeof(int32_t) * cd->buff_size_);
+	memset(cd->buf_in, 0, sizeof(int32_t) * cd->buff_size_);
+	memset(cd->buf_out, 0, sizeof(int32_t) * cd->buff_size_);
+	cd->gain_state_ = 0;
+
+	comp_set_state(dev, COMP_TRIGGER_RESET);
+
+	return ret;
+}
+
+static int aria_trigger(struct comp_dev *dev, int cmd)
+{
+	comp_info(dev, "aria_trigger()");
+
+	return comp_set_state(dev, cmd);
+}
+
+static int aria_copy(struct comp_dev *dev)
+{
+	struct comp_copy_limits c;
+	struct comp_buffer *source;
+	struct comp_buffer *sink;
+	uint32_t source_bytes;
+	uint32_t sink_bytes;
+	uint32_t buff_frag = 0;
+	int32_t *destp;
+	int32_t i, channel;
+	struct aria_data *cd = comp_get_drvdata(dev);
```

Comment:
lets keep all structs together in reverse xmas tree style followed by the other variables